src/beignet/_newton_raphson.py

```diff
@@ -0,0 +1,172 @@
+import warnings
+from typing import Any, Callable, Optional, Union
+
+import torch
+from torch import Tensor
+
+# Define function signature types
+FuncType = Callable[[Tensor, Tensor], tuple[Tensor, Any]]
```

Comment:
What is the `Any` in the return type?

Reply:
additional args you may want to pass to FPrime, e.g. in the test case, class probabilities